pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -372,31 +379,16 @@ private List<String> findCommonBrokerInstances(Set<String> brokerTenants) {
 
   private StreamingOutput sendRequestToBroker(String query, String instanceId, String traceEnabled, String queryOptions,
       HttpHeaders httpHeaders) {
-    InstanceConfig instanceConfig = _pinotHelixResourceManager.getHelixInstanceConfig(instanceId);
-    if (instanceConfig == null) {
-      LOGGER.error("Instance {} not found", instanceId);
-      throw QueryErrorCode.INTERNAL.asException();
-    }
-
-    String hostName = instanceConfig.getHostName();
-    // Backward-compatible with legacy hostname of format 'Broker_<hostname>'
-    if (hostName.startsWith(CommonConstants.Helix.PREFIX_OF_BROKER_INSTANCE)) {
-      hostName = hostName.substring(CommonConstants.Helix.BROKER_INSTANCE_PREFIX_LENGTH);
-    }
-
+    String hostName = getHost(instanceId);
     String protocol = _controllerConf.getControllerBrokerProtocol();
-    int port = _controllerConf.getControllerBrokerPortOverride() > 0 ? _controllerConf.getControllerBrokerPortOverride()
-        : Integer.parseInt(instanceConfig.getPort());
+    int port = getPort(instanceId);
```

Comment:
nit: this doubles the number of getHelixInstanceConfig calls. Maybe we can make one call to get instance config, and then pass it on for hostname and port extraction?